driver-jsdom/index.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const fs = require('fs/promises');
+const path = require('path');
+const jsdom = require('jsdom');
+
+exports.mochifyDriver = mochifyDriver;
+
+async function mochifyDriver(options = {}) {
+  options = Object.assign(
+    {
+      url: path.resolve(__dirname, './index.html'),
+      origin: 'http://localhost'
+    },
+    options
+  );
+  const stderr = options.stderr || process.stderr;
+
+  const host_document = await fs.readFile(options.url);
+  const virtual_console = new jsdom.VirtualConsole();
+
+  const { window } = new jsdom.JSDOM(host_document, {
+    runScripts: 'dangerously',
+    url: options.origin,
+    virtualConsole: virtual_console
+  });
+
+  function end() {
+    return null;
+  }
+
+  virtual_console.on('jsdomError', (error) => {
```

Comment:
This is kind of tricky: `jsdomError` will be triggered on general evaluation or runtime errors (which is why we want this), but _also_ for uncaught exceptions in the code it is running, in which case we currently print it twice as it's being handled by the client's `window.onerror` handler as well as this handler here.

Reply:
Does it throw a different type of error or is there anything on the error object that we can use identify this case?

Reply:
They indeed have a `type` property we can switch on. Good idea.